### ISSUE TITLE
Nuget pack can't find  dependencies if solution not above project

### ISSUE THREAD

**Issue by pimboden** [NuGet/Home#5316]
Hello 
I am using nuget version 4.1.0
I cant create a nuget package if the solution is not stored hierarchically higher than the project, and the projetc uses other nugets

My company has a weird file structure.
The solutions  are stored in a folder  "/solutions"
the projects are stored in  folders like "/libs", "/Web","/services" at the same level as the "/solutions" folder.

For example if I have a solution "IoC" I get this folder structure 

    /solutions/Ioc/ioc.sln
    /solutions/Ioc/packages /  <-- here the nuget packages are stored

    /libs/Ioc/Ioc/ioc.csproj
    /libs/Ioc/Ioc/packages.config
    /libs/Ioc/Ioc/IoC.nuspec

    /libs/Ioc/Ioc.tests/Ioc.tests.csproj

My project uses 2 nugets that should then be added as dependencies

my nuspec look like this 

    <?xml version="1.0"?>
    <package >
      <metadata>
        <id>$id$</id>
        <version>$version$</version>
        <title>$title$</title>
        <authors>IMP FIM</authors>
        <owners>IMP FIM</owners>
        <requireLicenseAcceptance>false</requireLicenseAcceptance>
        <description>Inversion of control package.</description>
        <releaseNotes>First release</releaseNotes>
        <copyright>Copyright 2017</copyright>
        <tags>IoC Dependeny Injection</tags>
      </metadata>
    </package>

If now I try to create a package  with 

      nuget pack IoC.csproj

I get  following output:

> Attempting to build package from 'IoC.csproj'.
>MSBuild auto-detection: using msbuild version '14.0' from 'C:\Program Files (x86)\MSBuild\14.0\bin\amd64'
>Packing files from 'C:\Code\Libs\IoC\IoC\bin\Debug'
>Using 'IoC.nuspec' for metadata.
>Found packages.config. Using packages listed as dependencies
>Unable to find 'CommonServiceLocator.1.3.nupkg'. Make sure the project has been built.

This of course is so, because Visual studio stored these files under the solution's directory.

I haven't found any documentation how to tell nuget pack, where too lookup for the referenced packages.

**Comment by thessalianpine** [comment]
I'm running into the same issue trying to get a solution to build/pack on our CI server. I'm using nuget 4.1.0.2450. The folder layout for our solution is also somewhat unusual with the .sln in the top-level directory and several projects located in various subfolders.

Example
```
basedir\solution.sln
basedir\sub1\sub2\projA\projA.csproj
basedir\sub1\sub2\projB\projB.csproj
basedir\sub1\projC\projC.csproj

The global package directory is at
basedir\packages

```
etc.

Each of the individual projects has one or more nuget packages it depends on. All packages are stored in the global packages directory and msbuild can find everything just fine, building from within VS and from the command line.

But no matter which project I try to pack, nuget fails to find the packages referenced by the project and stops the pack step with the same error

>Attempting to build package from 'MyProject.csproj'.
>MSBuild auto-detection: using msbuild version '12.0' from 'C:\Program Files (x86)\MSBuild\12.0\bin\amd64'.
>Packing files from 'C:\dev\AppProject\basedir\sub1\projC\bin'.
>Using 'MyProject.nuspec' for metadata.
>Found packages.config. Using packages listed as dependencies
>Unable to find 'EntityFramework.6.1.3.nupkg'. Make sure the project has been built.

The packages are available, packages.config, nuspec and hintpaths in the csproj look good.
I've tried putting the required .nupkg files in a different folder to see if nuget will pick them up, without success.

It would be helpful to be able to give nuget a hint where to look for packages if it can't find them on its own.

**Comment by lukepuplett** [comment]
Setup a post-build event with this:

`xcopy /HIDE "$(SolutionDir)\packages\*.*" "$(MSBuildProjectDirectory)\packages"`

Adjust paths as necessary. It'll copy the packages folder to the place where it can find it. Note that `repositoryPath` NuGet.config setting doesn't work - not for me, anyway.

**Comment by mishra14** [comment]
pinging @rohit21agrawal

**Comment by bjsafdie** [comment]
Same issue trying to set up a TeamCity build to do the build and pack.  I tried adding sub-dependencies directly to the nuspec file and tried the xcopy post build event as shown by lukepuplett.  No joy.  When nuget pack finds a packages.config file in the project, it tries to use it to add dependencies.  We need a command parameter to help NuGet find the directory for packages listed in packages.config -- something like the restore -PackagesDirectory option, but as a source for referenced packages.

**Comment by wmlrose** [comment]
I had a similar problem with nuget 4.3 packaging from a VS 2017 Post-build event. Don't know how good this solution is but inserting the following before the pack command worked for my purposes:

`nuget.exe config -Set repositoryPath="$(SolutionDir)..\<SomePathRelativeToSolutionFolder>\packages"`

**Comment by mishra14** [comment]
@rohit21agrawal can you take a peek to prioritize this issue?

**Comment by benjamin-devuyst** [comment]
I had similar problem, and this works for me : 
1. add a nuget.config in solution folder (this is where nuget.exe is invoked),
2. add following repositoryPath in it :
  <config>
    <add key="repositoryPath" value=".\packages" />
  </config>

this is strange, but it give nuget.exe a 'base path' to it self...

**Comment by bkwdesign** [comment]
Yes, thank you @foxrough - creating a `nuget.config` did the trick.
Here's what I chose to put in mine:
```
<?xml version="1.0" encoding="utf-8"?>
<configuration>
	<config>
	    <add key="dependencyVersion" value="Highest" />
	    <add key="globalPackagesFolder" value="C:\Users\myself\.nuget\packages" />
	    <add key="repositoryPath" value=".\packages" />
	</config>
</configuration>
```

IDK that the `globalPacakgesFolder` was really necessary, but, I just threw it in there. This technique allowed me to run a command where my nuget.exe was a directory higher from where the 'packages' were located. My `packages` were still higher up than the solution, which was yet higher up from where the project was located.

For example:
**C:\AzureApps\Adal>**`..\nuget.exe pack ".\trunk\MyProject\MyProjectApp.csproj" -Build`

So, fwiw, I ran the above command from the directory which contained both the `/packages` dir, and the `nuget.config` file

**Comment by WTRipper** [comment]
Using Process Monitor I saw that nuget.exe is looking in '%APPDATA%\NuGet\packages' as a fallback.  You can place the needed nugets there as a work around.
